### PR TITLE
Replace existing login flow with Google OAuth 2 integration

### PR DESCRIPTION
**Pull request by jtwillig**
# Task
Hook up OAuth / set up Google Cloud Project to allow authentication for Survey
https://incredihire-academy.atlassian.net/browse/SR-118

# Author Tasks

## Acceptance Criteria
- [X] N/A

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -39,3 +39,8 @@ DOCKER_IMAGE_BACKEND=survey_backend
 # Other config variables
 DOCKER_IMAGE_CELERYWORKER=survey_celeryworker
 TAG=latest
+
+GOOGLE_AUTHORIZATION_URL=https://accounts.google.com/o/oauth2/v2/auth,
+GOOGLE_TOKEN_URL=https://oauth2.googleapis.com/token
+GOOGLE_CLIENT_ID=233290826498-mfs7tcq5k1vdl2uhmavbcqn2k47dvk2t.apps.googleusercontent.com
```
Comment (Eugene-Polonsky) [inline .env.example:45]: hmm is this something we want committed to a repo?

**Review by Eugene-Polonsky — COMMENTED** [review]
Overall looks good but need unit tests for both frontend and backend. I'm thinking the actual oauth can be mocked out, but the functions calling the flows ought to be tested.

File: .env.example (hunk 1 above)
Comment (jtwillig) [reply .env.example:45]: Good call. This is addressed in my next commit. I have removed these URLs from the config entirely and I removed the value for GOOGLE_CLIENT_ID.

File: backend/app/api/routes/auth.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+GOOGLE_CLIENT_ID = settings.GOOGLE_CLIENT_ID
+GOOGLE_CLIENT_SECRET = settings.GOOGLE_CLIENT_SECRET
+
+
+@router.get("/login")
+async def login(state: str, request: Request):
+    redirect_uri = request.url_for("auth_callback")
+    google_auth_url = f"{settings.GOOGLE_AUTHORIZATION_URL}?client_id={GOOGLE_CLIENT_ID}&redirect_uri={redirect_uri}&response_type=code&scope=openid email profile&state={state}"
+    return RedirectResponse(url=google_auth_url)
```
[the remark this answers is not in the document]
Comment (jtwillig) [reply backend/app/api/routes/auth.py:27]: I have made changes to validate the redirect url in my next commit.

File: backend/app/api/routes/auth.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                ),
+            )
+        if not user.is_active:
+            raise HTTPException(status_code=400, detail="Inactive user")
+        response = RedirectResponse(url=state)
+        access_token = token_response.get("access_token")
+        expires_in = token_response.get("expires_in")
+        request.session["access_token"] = access_token
+        request.session["email"] = email
+        response.set_cookie("access_token", access_token, expires_in)
```
[the remark this answers is not in the document]
Comment (jtwillig) [reply backend/app/api/routes/auth.py:69]: I will make the cookie secure when referrer base is not http://localhost.

**Review by Eugene-Polonsky — APPROVED** [review]

**Review by Eugene-Polonsky — APPROVED** [review]

**Pull request merged**